ucb-rit/coldfront
# Load LRC data; support both instances with one branch

Fixes #324, #329, #331

**Changes**
- Back End
    - Added a feature flag `BRC_ONLY` to enable/disable BRC-specific features at runtime.
        - Enabled this flag when running setup commands in the test suite.
    - Modified `get_project_compute_resource_name` and `get_project_compute_allocation` to use feature flags, and removed duplicate methods.
    - Management Commands
        - Renamed and modified `add_accounting_defaults` to create `Resources` and `AllocationAttributeTypes` based on feature flags.
        - Added `load_lrc_data` for loading LRC user, project, allocation, and billing data from various data sources.
- Front End
    - On the home page, merged project lists (by cluster) into one table.
    - (LRC only) On the Allocation Detail view, displayed fully-formed billing IDs for the `Allocation` as well as for individual `AllocationUsers`.
    - (LRC only) On the User Profile, displayed the fully-formed billing ID for the monthly user account fee.
- Deployment
    - Updated the Ansible and other bootstrap configuration so that both deployments can be supported from one branch.
        - Removed hardcoded references to the BRC instance, replacing them with variables/inputs.
        - Minimized differences between the Ansible setups for development vs. staging/production.
- Miscellanea
    - Modified various test modules to inherit from `TestBase` to minimize duplication of setup command calls.
    - Added notes about MyLRC and reordered installation steps in `README.md`.

**How to Test**
- Review the code changes and add comments as needed.
- Ensure that the test suite passes on Jenkins.
- Manual Testing
    - Create flags.
        - In the Django admin interface, click on "Flag states".
        - "ADD FLAG STATE": Select flag `BRC_ONLY`, condition "boolean", value `True`. Make the state required.
        - "ADD FLAG STATE": Select flag `LRC_ONLY`, condition "boolean", value `False`. Make the state required.
        - In general, both should not be enabled at the same time.
    - Make a backup of `main.yml`. Modify it to include the changes to `development/main.copyme`.
    - Re-run the Ansible playbook.
    - Ensure that, when `LRC_ONLY` is enabled, billing-related data appears on the User Profile and on the Allocation Detail view; when it is disabled, that data should not appear.

**Notes**
- The use of the `BRC_ONLY` flag in the test suite should be replaced with a longer-term solution that allows for testing of both instances.
- Many locations in the front-end still use hardcoded strings like "BRC", "MyBRC", and "Berkeley Research Computing"; these need to be changed into variables.
